lib/ssl/src/dtls_connection.erl

```diff
@@ -1107,9 +1117,9 @@ send_handshake_flight(#state{static_env = #static_env{socket = Socket,
 			     connection_states = ConnectionStates0,
                              ssl_options = #{log_level := LogLevel}} = State0,
                       Epoch) ->
-    %% TODO remove hardcoded Max size
+    #{current_write := #{max_fragment_length := MaxFragmentLength}} = ConnectionStates0,
     {Encoded, ConnectionStates} =
-	encode_handshake_flight(lists:reverse(Flight), Version, 1400, Epoch, ConnectionStates0),
+	encode_handshake_flight(lists:reverse(Flight), Version, MaxFragmentLength, Epoch, ConnectionStates0),
```

Comment:
As I see it, encode_handshake_flight/5 calls dtls_handshake:fragment_handshake/2 with the plaintext (Flight) and MaxFragmentLength. So the plaintext is being broken up into chunks not larger than MaxFragmentLength.
dtls_record:encode_handshake/4 handles the ciphering so the final packet is generated there which will be larger than the hardcoded 1400.
I agree the hardcoding the MTU is a problem but the code does what it is supposed to with the MaxFragmentLength operating on the plaintext.
If you prefer I can make this max(1400, MaxFragmentLength) to keep the new option from making the DTLS plaintext larger than what would be generated with the existing code.

Reply:
I have some thoughts here that the hardcoded size was never really applied. Maybe, we should hardcode it to `?MAX_PLAIN_TEXT_LENGTH` to get the original behavior? @IngelaAndin

Reply:
So I have been thinking about this, re reading the DTLS RFC. As I remember the hard coded 1400 was chosen to be a little smaller than than expected "common case"  MTU (of course the tm correct thing to do is to have a discovery mechanism that we have not had time to implement so far, at the leas we should make it configurable), so that the encrypted DTLS record holding a handshake fragment normally should fit into one datagram, as this is the requirement of the RFC.  So in my mind it would it not be the min(1400, MaxFragmentLen) that we want?!

Reply:
Then I think we can resolve this item, too, as it correctly sets it to 1400.